src/sentry/search/snuba/executors.py

```diff
@@ -564,3 +569,244 @@ def calculate_hits(
+                # Since we have GroupEnvironment data that persists beyond retention, whereas Snuba only keeps events in the retention period.
+                return None, None
+            else:
+                table_alias = "groups."
+
+            if isinstance(converted_filter[0], list):
+                converted_filter[0][1][0] = table_alias + converted_filter[0][1][0]
+            else:
+                converted_filter[0] = table_alias + converted_filter[0]
+
+        # if search_filter.key.name == "first_release":
+        #     # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+        #     release = Release.objects.filter(
+        #         version=converted_filter[2],
+        #         organization_id=Project.objects.get(id=project_ids[0]).organization_id,
+        #     )
+        #     if not release:
+        #         # TODO: This means there will be no results and we do not need to run this query!
+        #         # Temp solution is to set value to -1, which should never be a real release ID.
+        #         converted_filter[2] = -1
+        #     else:
+        #         converted_filter[2] = release[0].id
+
+        return converted_filter
+
+    def query(
+        self,
+        projects,
+        retention_window_start,
+        group_queryset,
+        environments,
+        sort_by,
+        limit,
+        cursor,
+        count_hits,
+        paginator_options,
+        search_filters,
+        date_from,
+        date_to,
+    ):
+
+        now = timezone.now()
+        end = None
+        end_params = filter(None, [date_to, get_search_filter(search_filters, "date", "<")])
+        if end_params:
+            end = min(end_params)
+
+        if not end:
+            end = now + ALLOWED_FUTURE_DELTA
+
+        retention_date = max(filter(None, [retention_window_start, now - timedelta(days=90)]))
+
+        start_params = [date_from, retention_date, get_search_filter(search_filters, "date", ">")]
+        start = max(filter(None, start_params))
+
+        end = max([retention_date, end])
+
+        if start == retention_date and end == retention_date:
+            return self.empty_result
+
+        if start >= end:
+            return self.empty_result
+
+        for i in range(len(search_filters)):
+            search_filter = search_filters[i]
+            if search_filter.key.name == "first_release":
+                # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+                release = Release.objects.filter(
+                    version=search_filter[2].value, organization_id=projects[0].organization_id
+                )
+                if not release:
+                    return self.empty_result
+                else:
+                    search_filters[i] = SearchFilter(
+                        SearchKey("first_release"), "=", SearchValue(release[0].id)
```

Comment:
You can make this more readable by running a map function over search_filter  instead of iterating with an index. That way  you would not have the index around.
Also I guess you need to add the table prefix.